test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -298,6 +306,78 @@ instance (KnownSymbol param, HasPath sub) => HasPath (Capture param t :> sub)
         let (verb, sub) = getPath (Proxy @sub)
         in (verb, "/{" <> symbolVal (Proxy :: Proxy param) <> "}" <> sub)
 
+
+
+{-------------------------------------------------------------------------------
+               roundtripAndGolden for an entire Api type
+-------------------------------------------------------------------------------}
+
+aesonTestCase :: Checkable a => Proxy a -> AesonTestCase
+aesonTestCase proxy =
+    AesonTestCase
+    { typeRep = Data.Typeable.typeRep proxy
+    , typeSpec = roundtripAndGolden proxy
+    }
+
+data AesonTestCase = AesonTestCase
+    { typeRep :: TypeRep
+    , typeSpec :: Spec
+    }
+
+instance Eq AesonTestCase where
+    a == b = (typeRep a) == (typeRep b)
+
+instance Ord AesonTestCase where
+    compare a b = compare (typeRep a) (typeRep b)
+
+class JsonTestable api where
+    roundtripAndGoldenPerType :: Proxy api -> Set AesonTestCase
+
+type Checkable a = (Typeable a, Arbitrary a, ToJSON a, FromJSON a)
+
+-- | Run tests on the return-type
+instance {-# OVERLAPS #-} (Checkable a, Method m)
+    => JsonTestable (Verb m s ct a) where
+    roundtripAndGoldenPerType _ =
+        Set.singleton $ aesonTestCase $ Proxy @a
+
+instance {-# OVERLAPS #-} (Checkable a, Method m)
+    => JsonTestable (Verb m s ct [a]) where
+    roundtripAndGoldenPerType _ =
+        Set.singleton $ aesonTestCase $ Proxy @a
+
+-- | Run tests on captured types
+instance (JsonTestable b, JsonTestable t, KnownSymbol param)
+    => JsonTestable (Capture param t :> b) where
+    roundtripAndGoldenPerType _ =
+        Set.union
+            (roundtripAndGoldenPerType $ Proxy @t)
+            (roundtripAndGoldenPerType $ Proxy @b)
```

Comment:
That one is incorrect. There's no reason for a param `t` to have a a JSON instance. Instead, we should be checking for `HttpApiData` roundtrips here!